### DISCUSSION TITLE
Allow user to dismiss the habit for one day

### DISCUSSION THREAD

**Discussion by consoleaf** [iSoron/uhabits#272] [Feature Requests]
What about adding a button for hiding a task for one day?

**Comment by iSoron** [comment]
Thank you for the suggestion, @Fantaser99. I think this would be an interesting feature. Instead of adding button, we could let the user swipe the habit away, on the main screen.

**Comment by jt007** [comment]
Really looking forward to this @iSoron ! Do you have a time frame for when this could be added?

**Comment by StanczakDominik** [comment]
I suspect there's nobody working on this right now. I haven't really done any Android dev work before, but I'd really like to have this functionality in there somewhere. I may find a few hours within the next month to start on this.

The way I see this working is:

* Add a third hideable state for habits called "skipped", off by default
* Add some handler to toggle this state for a habit on side swipe on the habit's name. 
* Maybe add some visual feedback, say, a "Habit skipped" pop-up
* Add a menu entry for showing/hiding skipped habits just like completed and archived ones work right now
* Possibly add a button to skip or unskip multiple habits? This would probably be easy to do as a follow-up, and I'd rather not bother as a beginner. Swiping seems sufficient
* The crux, and I'm not sure yet what good practices are for this kind of thing: at every day start, for every skipped habit, if it would be due today, unskip it.

**Comment by iSoron** [comment]
@StanczakDominik There isn't anyone working on this feature right now. Pull requests would be very welcome. As a first implementation, I would recommend simply making the habits temporarily swipeable on the main screen; after restarting the app, they would all appear again. Later, we can see how to add persistence to this. Please feel free to submit a work-in-progress pull request if you would like to discuss the implementation details.

**Comment by lashoun** [comment]
If I may, I think the system used in [this app](https://everyday.app) is quite nice. It's a third state which keeps a streak alive as long as you don't skip the habit twice in a row. As of now when you press on a habit it either ticks or unticks it: we could make it so that states rotate, for example unticked -> ticked -> skipped -> unticked. For ease of implementation, the "don't skip twice in a row" rule can be left unenforced by the app itself, it'd be more of a personal rule. The skipped state would therefore be equivalent to the ticked state for the app, the only difference being that it is displayed differently (by emptying the diagonal half of the square for instance).

**Comment by iSoron** [comment]
@lashoun The feature you describe is being discussed in #88. Essentially, skipping days is an essential feature for other don't-break-the-chain apps because one failure in those apps resets your progress to zero. With our scoring algorithm, however, occasional failures have almost zero impact anyway.

The feature being discussed in this issue is simply hiding the habit until the following day, so that the list becomes less cluttered with tasks that you know you can no longer complete. It will still be recorded as a lapse.

**Comment by scrimbly** [comment]
@StanczakDominik Did you make any progress on this? I'm craving this feature, and I'm motivated to code it up. Any code/prototypes/sketches you'd like to share?

**Comment by StanczakDominik** [comment]
I get the same cravings, but unfortunately I haven't been able to start on it yet due to deadlines on other projects... If you'd like to start, go right ahead, and I'll try to help however I can.

**Comment by scrimbly** [comment]
Poll: Swipe left to dismiss or swipe right to dismiss? Or both/either?

**Comment by StanczakDominik** [comment]
I'd say both!

**Comment by scrimbly** [comment]
I just posted a WIP where I got the swiping + dismissing + filtering working, but I'm still working on properly refreshing the list after the swipe.

**Comment by jt007** [comment]
@iSoron @scrimbly @StanczakDominik 

Hi, any news on this feature?  Would be really helpful to be able to remove tasks that are not possible to do today, to make the app less cluttered on busy days. This will make it much easier to really Focus on what's important! 

Some of the options to make it possible:
   - swipe the task away         or
   - long press the item. Then click on the three dots up in the right corner (a meny with Archive and Delete pops up). Just add 
          "dismiss today" to this menu. Should be an easy fix.
 
Has taken more than three years now. Is it possible to get this feature implemented? Will make the app 10 times better!

Thank you very much for your help :-)

**Comment by KristianTashkov** [comment]
@jt007 I built skip/fail days in https://github.com/iSoron/uhabits/pull/610 and have been using it on my phone for a month now. The PR is huge though so it still doesn't have a review and isn't merged. If you want you can build yourself the APK from that branch or you can contact me so I can give you the one I use.
